plugins/services/src/js/utils/PodUtil.js

```diff
@@ -90,22 +90,53 @@ var PodUtil = {
 
       // Filter combined container list to remove potential duplicates
       const containerIds = new Map();
-      combinedContainers = combinedContainers.filter(function({ containerId }) {
-        if (containerId != null && !containerIds.has(containerId)) {
-          containerIds.set(containerId);
+      const nullIdContainers = [];
+      combinedContainers = combinedContainers.filter(function(container) {
+        if (
+          container.containerId != null &&
+          !containerIds.has(container.containerId)
+        ) {
+          containerIds.set(container.containerId);
 
           return true;
         }
 
+        if (container.containerId == null) {
+          nullIdContainers.push(container);
```

Comment:
🎨 This seems to be sorting a list into two lists, I would rather like to see side-effect free filters with just a negative predicate if it's really distinct or another predicate.